src/walk.rs

```diff
@@ -0,0 +1,421 @@
+//! Graph traversal module providing iterator-based walking of AST graphs.
+//!
+//! This module implements a depth-first traversal mechanism for graph structures
```

Comment:
I think we actually have breadth-first implementation
```suggestion
//! This module implements a breadth-first traversal mechanism for graph structures
```